test/minikube/minikube_base_tde_test.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func applyStoragePasswordSecret(t *testing.T, namespaceName string, name string, passwords []string) {
+	assert.Greater(t, len(passwords), 0)
+	opts := k8s.NewKubectlOptions("", "", namespaceName)
+	var kubectlArgs []string
+	kubectlArgs = append(kubectlArgs, "create", "secret", "generic", name, "--dry-run=client", "-o", "yaml")
+	kubectlArgs = append(kubectlArgs, "--from-literal", fmt.Sprintf("target=%s", passwords[0]))
+	for i, password := range passwords[1:] {
+		kubectlArgs = append(kubectlArgs, "--from-literal", fmt.Sprintf("historical-%d=%s", i, password))
+	}
+	secret, err := k8s.RunKubectlAndGetOutputE(t, opts, kubectlArgs...)
+	require.NoError(t, err)
+	tmpfile, err := ioutil.TempFile("", "tde_secret")
+	require.NoError(t, err)
+	defer os.Remove(tmpfile.Name())
+	_, err = tmpfile.WriteString(secret)
+	require.NoError(t, err)
+	err = tmpfile.Close()
+	require.NoError(t, err)
+	k8s.RunKubectl(t, opts, "apply", "-f", tmpfile.Name())
+	testlib.AddTeardown(testlib.TEARDOWN_ADMIN, func() {
+		k8s.RunKubectlE(t, opts, "delete", "secret", name)
+	})
+}
+
+func TestAdminColdStartWithTDE(t *testing.T) {
+	// TODO: remove this whenever the image tested in nuodb-helm-charts CI
+	// supports 'tde_monitor' service, i.e. whenever the version
+	// is bumped to >4.1.1
+	if os.Getenv("NUODB_DEV") != "true" {
+		t.Skip("'tde_monitor' service is not supported in released versions")
+	}
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	options := helm.Options{
+		SetValues: map[string]string{
+			"admin.tde.secrets.demo":                "demo-tde-secret",
+			"database.name":                         "demo",
+			"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"admin.tde.storagePasswordsDir":         "/etc/nuodb/encryption",
+		},
+	}
+	randomSuffix := strings.ToLower(random.UniqueId())
+	namespaceName := fmt.Sprintf("admin-cold-start-with-tde-%s", randomSuffix)
+	testlib.CreateNamespace(t, namespaceName)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	options.KubectlOptions = kubectlOptions
+
+	password := strings.ToLower(random.UniqueId())
+	applyStoragePasswordSecret(t, namespaceName, options.SetValues["admin.tde.secrets.demo"], []string{password})
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+	helmChartReleaseName, _ := testlib.StartAdmin(t, &options, 1, namespaceName)
+	admin0 := fmt.Sprintf("%s-nuodb-cluster0-0", helmChartReleaseName)
+	testlib.Await(t, func() bool {
+		return testlib.GetStringOccurrenceInLog(t, namespaceName, admin0,
+			"Successfully updated storage passwords for dbName=demo", &corev1.PodLogOptions{}) >= 1
+	}, 30*time.Second)
+
```

Comment:
Can you assert that `nuocmd check data-encryption --password <the password>` is successful. You could also specify `--timeout` and eliminate the need for polling in the test.

Reply:
Okay, I will add a one-shot `check data-encryption` to assert that the password is set.
Unfortunately, we don't support `--timeout` with this command. Certainly, something that we could add.